packages/desktop-client/src/components/reports/reports/CashFlowCard.tsx

```diff
@@ -83,6 +93,153 @@ function CustomLabel({
+) {
+  return (
+    <BarChart
+      width={width}
+      height={height}
+      data={[
+        {
+          income,
+          expenses,
+        },
+      ]}
+      margin={{
+        top: 10,
+        bottom: 0,
+      }}
+    >
+      <Bar dataKey="income" fill={chartTheme.colors.blue} barSize={14}>
+        <LabelList
+          dataKey="income"
+          position="left"
+          content={<CustomLabel name={t('Income')} />}
+        />
+      </Bar>
+
+      <Bar dataKey="expenses" fill={chartTheme.colors.red} barSize={14}>
+        <LabelList
+          dataKey="expenses"
+          position="right"
+          content={<CustomLabel name={t('Expenses')} />}
+        />
+      </Bar>
+    </BarChart>
+  );
+}
+
+function retViewDetailed(
+  totalIncome: number,
+  totalExpenses: number,
+  totalTransfers: number,
+  isCardHovered: boolean,
+) {
+  return (
+    <View
+      style={{
+        paddingTop: 20,
+        alignItems: 'flex-end',
+        color: theme.pageText,
+      }}
+    >
+      <AlignedText
+        style={{ marginBottom: 5, minWidth: 160 }}
+        left={
+          <Block>
+            <Trans>Income:</Trans>
+          </Block>
+        }
+        right={
+          <Text style={{ fontWeight: 600 }}>
+            <PrivacyFilter>{integerToCurrency(totalIncome)}</PrivacyFilter>
+          </Text>
+        }
+      />
+
+      <AlignedText
+        style={{ marginBottom: 5, minWidth: 160 }}
+        left={
+          <Block>
+            <Trans>Expenses:</Trans>
+          </Block>
+        }
+        right={
+          <Text style={{ fontWeight: 600 }}>
+            <PrivacyFilter>{integerToCurrency(totalExpenses)}</PrivacyFilter>
+          </Text>
+        }
+      />
+
+      <AlignedText
+        style={{ marginBottom: 5, minWidth: 160 }}
+        left={
+          <Block>
+            <Trans>Transfers:</Trans>
+          </Block>
+        }
+        right={
+          <Text style={{ fontWeight: 600 }}>
+            <PrivacyFilter>{integerToCurrency(totalTransfers)}</PrivacyFilter>
+          </Text>
+        }
+      />
+      <Text style={{ fontWeight: 600 }}>
+        <PrivacyFilter activationFilters={[!isCardHovered]}>
+          <Change amount={totalIncome + totalExpenses + totalTransfers} />
+        </PrivacyFilter>
+      </Text>
+    </View>
+  );
+}
+
+function retChartDetailed(
+  graphData: {
+    expenses: { x: Date; y: number }[];
+    income: { x: Date; y: number }[];
+    balances: { x: Date; y: number }[];
+    transfers: { x: Date; y: number }[];
+  },
+  isConcise: boolean,
+  height: number,
+) {
+  if (height < 290) {
+    return (
+      <Warning style={{ paddingTop: 5, paddingBottom: 5 }}>
+        Additional widget height required to display chart, edit dashboard to
+        increase widget height
+      </Warning>
+    );
+  } else {
+    return (
+      <CashFlowGraph
+        graphData={graphData}
+        isConcise={isConcise}
+        showBalance={true}
+      />
+    );
+  }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏unnecessary nesting. Prefer to return early.
```suggestion
  }
  
  return (
      <CashFlowGraph
        graphData={graphData}
        isConcise={isConcise}
        showBalance={true}
      />
    );
```